087-external-certificate-manager.md

```diff
@@ -0,0 +1,324 @@
+At a high level:
+* Strimzi will use cert path validation using the `java.security` libraries to determine whether a new Cluster CA public cert was signed with a new private key.
+* If a new private key was used, Strimzi will increment both the `strimzi.io/ca-key-generation` and `strimzi.io/ca-cert-generation` annotations on the Cluster CA cert Secret and roll the Kafka pods from CaReconciler to trust the new public cert.
+* If the private key has not changed, Strimzi will increment only the `strimzi.io/ca-cert-generation` annotation on the Cluster CA cert Secret and roll the Kafka pods from KafkaReconciler to trust the new public cert.
+* If a new end-entity certificate is issued, Strimzi will use cert path validation to verify it is trusted by the latest Cluster CA cert Secret before using it.
+* Strimzi will update the `strimzi.io/cluster-ca-cert-generation` annotation on the Kafka pods to match the annotation on the Cluster CA cert Secret when it copies over a new end-entity certificate
+
+#### Handling Cluster CA trust rollout
+
+When a new Kafka cluster is created Strimzi will copy the CA public cert from the Secret identified in `clusterCa.publicCert` to the `<CLUSTER_NAME>-cluster-ca-cert` Secret in a data entry named `ca.crt`.
+Strimzi will add three annotations to the `<CLUSTER_NAME>-cluster-ca-cert` Secret:
+* `strimzi.io/ca-cert-hash` with the value being a hash of the certificate.
+* `strimzi.io/ca-key-generation` initially set to 0.
+* `strimzi.io/ca-cert-generation` initially set to 0.
+
+During a reconciliation Strimzi will check the hash of the certificate stored in `clusterCa.publicCert` Secret to see if an update is needed.
+If the certificate has changed Strimzi will perform cert path validation using the `java.security` libraries to determine whether a new Cluster CA public cert was signed with a new private key.
+If the private key has not changes, Strimzi will copy over the new certificate, replacing the existing one.
+It will also update the `strimzi.io/ca-cert-hash` and increment the `strimzi.io/ca-cert-generation` annotation.
+Each of the component reconcilers will check the `strimzi.io/cluster-ca-cert-generation` on their pods during their reconcile loop, and update the pod annotation and roll the pods if the generation is out of date. 
+
+> ![Renewing the cluster CA public cert](./images/087-cert-renewals.png)
+>
+> Fig 1: Existing and proposed workflow when the user provides a new cluster CA public cert
+
+#### Issuing Cluster CA end-entity certificates
+
+When Strimzi needs to issue a certificate, instead of using the existing internal mechanism it will create a `Certificate` custom resource.
+Strimzi will specify the required CN/SANs in the `Certificate` resource for the end-entity certificate.
+Strimzi will set the `secretName` field in the `Certificate` resource as `<CERT_SECRET>-cm`, where `<CERT_SECRET>` is the name of the Secret Strimzi currently uses, for example `<CLUSTER_NAME>-cluster-operator-certs-cm`.
```

Comment:
can we have an snippet example of such `Certificate` custom resource created by the operator?

Reply:
@katheris maybe did you miss this comment in the last batch of fixes? or do you have reasons for not putting a snippet example of `Certificate` resource?

Reply:
@ppatierno yeah I did miss it, I've added the example now